### PR TITLE
feat: add Prometheus exec time histogram

### PR DESCRIPTION
**Pull request by jmbass**

### CODE REVIEW

**Comment by jmbass** [comment]
For reference, the output metric in the /metrics endpoint
```
# HELP go_graphkb_graph_query_execution_time The time execution in ms of queries.
# TYPE go_graphkb_graph_query_execution_time histogram
go_graphkb_graph_query_execution_time_bucket{le="1"} 0
go_graphkb_graph_query_execution_time_bucket{le="50"} 0
go_graphkb_graph_query_execution_time_bucket{le="75"} 1
go_graphkb_graph_query_execution_time_bucket{le="100"} 1
go_graphkb_graph_query_execution_time_bucket{le="500"} 4
go_graphkb_graph_query_execution_time_bucket{le="1000"} 4
go_graphkb_graph_query_execution_time_bucket{le="2500"} 4
go_graphkb_graph_query_execution_time_bucket{le="5000"} 4
go_graphkb_graph_query_execution_time_bucket{le="10000"} 4
go_graphkb_graph_query_execution_time_bucket{le="+Inf"} 4
go_graphkb_graph_query_execution_time_sum 608
go_graphkb_graph_query_execution_time_count 4
```

File: internal/knowledge/querier.go [hunk 1]
```diff
@@ -72,7 +73,12 @@ func (q *Querier) queryInternal(ctx context.Context, cypherQuery string) (*Queri
 		return nil, translation.Query, err
 	}
 
-	logrus.Debugf("Found results in %dms", s.Execution/time.Millisecond)
+	execution_time := s.Execution / time.Millisecond
+
+	logrus.Debugf("Found results in %dms", execution_time)
```
Comment (ShimmerGlass) [inline internal/knowledge/querier.go:78]:
```suggestion
	logrus.Debugf("Found results in %s", s.Execution)
```

File: internal/knowledge/querier.go [hunk 2]
```diff
@@ -72,7 +73,12 @@ func (q *Querier) queryInternal(ctx context.Context, cypherQuery string) (*Queri
 		return nil, translation.Query, err
 	}
 
-	logrus.Debugf("Found results in %dms", s.Execution/time.Millisecond)
+	execution_time := s.Execution / time.Millisecond
```
Comment (ShimmerGlass) [inline internal/knowledge/querier.go:76]: Vars should be in camelCase

File: internal/knowledge/querier.go [hunk 3]
```diff
@@ -72,7 +73,12 @@ func (q *Querier) queryInternal(ctx context.Context, cypherQuery string) (*Queri
...
+
+	logrus.Debugf("Found results in %dms", execution_time)
+
+	metrics.GraphQueryTimeExecution.
+		WithLabelValues().Observe(float64(execution_time))
```
Comment (ShimmerGlass) [inline internal/knowledge/querier.go:81]: you could use `s.Execution.Seconds() / 1000` (gives you a float, and prevents loosing definition from integer division)

File: internal/metrics/metrics.go [hunk 4]
```diff
@@ -11,6 +11,13 @@ var StartTimeGauge = promauto.NewGauge(prometheus.GaugeOpts{
 	Help: "The timestamp of the time when the app started",
 })
 
+// GraphQueryTimeExecution reports the time execution in ms for queries.
+var GraphQueryTimeExecution = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "go_graphkb_graph_query_execution_time",
+	Help:    "The time execution in ms of queries.",
+	Buckets: []float64{1, 50, 75, 100, 500, 1000, 2500, 5000, 10000},
```
Comment (ShimmerGlass) [inline internal/metrics/metrics.go:18]: Opt: use `prometheus.ExponentialBuckets()` to generate buckets

File: internal/knowledge/querier.go (hunk 3 above)
Comment (jmbass) [reply internal/knowledge/querier.go:81]: Good catch, thanks 
`s.Execution.Seconds() * 1000` for we want ms

File: internal/metrics/metrics.go (hunk 4 above)
Comment (jmbass) [reply internal/metrics/metrics.go:18]: I'll leave it like this for now as I'd like to see the frequency of the observations within these quantiles first.

File: internal/knowledge/querier.go [hunk 5]
```diff
@@ -72,7 +73,12 @@ func (q *Querier) queryInternal(ctx context.Context, cypherQuery string) (*Queri
 		return nil, translation.Query, err
 	}
 
-	logrus.Debugf("Found results in %dms", s.Execution/time.Millisecond)
+	executionTime := s.Execution.Seconds() * 1000
```
Comment (clems4ever) [inline internal/knowledge/querier.go:76]: why not simply use .Milliseconds() instead?

**Review by clems4ever — CHANGES_REQUESTED** [review]

File: internal/knowledge/querier.go (hunk 5 above)
Comment (clems4ever) [reply internal/knowledge/querier.go:76]: https://pkg.go.dev/time#Duration.Milliseconds

Comment (jmbass) [reply internal/knowledge/querier.go:76]: To avoid casting to float afterwards when using .Observe() , but it makes sense to use this instead.

**Review by clems4ever — APPROVED** [review]

**Review by ShimmerGlass — APPROVED** [review]

**Pull request merged**